spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTransactionManager.java

```diff
@@ -91,8 +95,9 @@ protected void doBegin(Object transactionObject, TransactionDefinition transacti
 							"TransactionDefinition.PROPAGATION_REQUIRED");
 		}
 		Tx tx = (Tx) transactionObject;
-		final TransactionContext targetTransactionContext = tx.transactionManager.begin();
 		if (transactionDefinition.isReadOnly()) {
+			final ReadContext targetTransactionContext = this.databaseClient
+					.readOnlyTransaction();
 
 			tx.transactionContext = new TransactionContext() {
```

Comment:
I thought we wouldn't need this whole wrapper below when using `readOnlyTransaction`??

Reply:
We need a wrapper because from Spring Data's   point of view both transactions are the same type. I am substituting in the read-only context and Spring Data is not supposed to be aware of the difference. This is why.